sys/trickle/trickle.c

```diff
@@ -33,38 +33,44 @@ void trickle_callback(trickle_t *trickle)
 
 void trickle_interval(trickle_t *trickle)
 {
+    assert(trickle->I > 0);
+
     uint32_t old_interval = trickle->I;
     uint32_t max_interval = trickle->Imin << trickle->Imax;
     uint32_t diff = old_interval - trickle->t;
 
     trickle->I *= 2;
-
-    if ((trickle->I == 0) || (trickle->I > max_interval)) {
+    if (trickle->I > max_interval) {
         trickle->I = max_interval;
         old_interval = max_interval / 2;
     }
 
     DEBUG("trickle: I == %" PRIu32 ", diff == %" PRIu32 "\n", trickle->I, diff);
 
     trickle->c = 0;
+    /* old_interval == trickle->I / 2 */
     trickle->t = random_uint32_range(old_interval, trickle->I);
 
     trickle->msg_time = (trickle->t + diff) * MS_PER_SEC;
     xtimer_set_msg64(&trickle->msg_timer, trickle->msg_time, &trickle->msg,
                      trickle->pid);
-
 }
 
 void trickle_reset_timer(trickle_t *trickle)
 {
-    trickle_stop(trickle);
-    trickle_start(trickle->pid, trickle, trickle->msg.type, trickle->Imin,
-                  trickle->Imax, trickle->k);
+    if (trickle->I > trickle->Imin) {
```

Comment:
actually looking again at the code, `trickle->I` can never be equal `I->Imin` after a `reset`, because `trickle_interval()` modifies it immediately before any other `trickle_reset()` can be called. (I am not talking about shared access from multiple threads, for this a mutual exclusion mechanism is needed here)